tests/conftest.py

```diff
@@ -1,19 +1,144 @@
 
 def pytest_addoption(parser):
     parser.addoption("--runslow", action="store_true", default=False, help="run slow tests")
+    parser.addoption("--run-dsi-tests", action="store_true", default=False, help="run DSI integration tests")
 
 
 def pytest_configure(config):
     config.addinivalue_line("markers", "slow: mark test as slow to run")
+    config.addinivalue_line("markers", "dsi: mark test as part of the DSI framework")
+
+    if config.getoption("--run-dsi-tests"):
+        logging.basicConfig(
+            level=logging.INFO,
+            format='%(asctime)s - %(levelname)s - %(message)s',
+            datefmt='%Y-%m-%d %H:%M:%S',
+            force=True
+        )
+        logging.info("--- DSI: Loading environment variables ---")
+        try:
+            dotenv_path = project_root / '.env'
+            if load_dotenv(dotenv_path=dotenv_path, override=True):
+                logging.info(f"DSI: Successfully loaded environment variables from: {dotenv_path}")
+            else:
+                logging.warning(f"DSI: Could not find .env file at {dotenv_path}. Using system variables.")
+        except Exception as e:
+            pytest.fail(f"DSI: A critical error occurred while loading the .env file: {e}", pytrace=False)
+        
+        logging.info("--- DSI: Test generation will be handled by pytest parametrization ---")
 
 
 def pytest_collection_modifyitems(config, items):
-    if config.getoption("--runslow"):
-        # --runslow given in cli: do not skip slow tests
+    if not config.getoption("--runslow"):
+        skip_slow = pytest.mark.skip(reason="need --runslow option to run")
+        for item in items:
+            if "slow" in item.keywords:
+                item.add_marker(skip_slow)
+
+    if not config.getoption("--run-dsi-tests"):
+        items[:] = [item for item in items if "dsi" not in item.keywords]
+
+
+def pytest_sessionfinish(session: Any, exitstatus: int) -> None:
+    if session.config.getoption("--run-dsi-tests"):
+        logging.info("--- DSI: Test session finished. No generated files to clean up. ---")
+
+
+# --- DSI Framework Fixtures with Improved Error Handling ---
+
+@pytest.fixture(scope="session")
+def mindsdb_server(pytestconfig) -> Generator[Any, None, None]:
+    if not pytestconfig.getoption("--run-dsi-tests"):
+        yield None
         return
-    skip_slow = pytest.mark.skip(reason="need --runslow option to run")
-    for item in items:
-        if "slow" in item.keywords:
-            item.add_marker(skip_slow)
+    
+    from tests.integration.handlers.utils.helpers import connect_to_mindsdb
+        
+    try:
+        server = connect_to_mindsdb()
+        yield server
+    except Exception as e:
+        pytest.fail(
+            f"DSI: Failed to connect to MindsDB via SDK. Please ensure MindsDB is running and accessible. Error: {e}", 
+            pytrace=False
+        )
+
+
+@pytest.fixture(scope="session")
+def query_logger(pytestconfig):
```

Comment:
it might be better to subclass or patch the SDK client so that it logs queries automatically. That way we don't have to remember to log each one separately.